recipe/meta.yaml

```diff
@@ -22,17 +22,18 @@ requirements:
     - wheel
   run:
     - python
+    - comm >=0.1.3
     - ipython >=6.1.0
-    - ipykernel >=4.5.1
     - traitlets >=4.3.1
-    - widgetsnbextension >=4.0.4,<5.0.0
-    - jupyterlab_widgets >=3.0.4,<4.0.0
+    - widgetsnbextension >=4.0.10,<4.1
+    - jupyterlab_widgets >=3.0.10,<3.1
```

Comment:
Where do these upper bounds come from because I'm not seeing any listed [here](https://github.com/jupyter-widgets/ipywidgets/blob/8.1.2/python/ipywidgets/setup.cfg#L40-L41)?

Reply:
it's because of `~=`
see: https://packaging.python.org/en/latest/specifications/version-specifiers/#compatible-release

Reply:
Thanks for sharing! That makes sense and I'm bookmarking it now. Lol